projects/ngrx.io/content/guide/effects/operators.md

```diff
@@ -0,0 +1,176 @@
+# Effects operators
+
+As part of Effect library, NgRx provides some useful operators that are frequently
+used. 
+
+
+## `ofType`
+
+The `ofType` operator filters the stream of actions based on either string
+values or Action Creators.
+
+The generic for the `Actions<TypeUnion>` must be provided in order for type 
+inference to work properly with string values. Action Creators that are based on
+`createAction` function do not have the same limitation.
+
+The `ofType` operator takes up to 5 arguments with proper type inference. It can
+take even more, however the type would be inferred as `Action` interface.
+
+<code-example header="auth.effects.ts">
+import { Injectable } from '@angular/core';
+import { Actions, ofType, createEffect } from '@ngrx/effects';
+import { of } from 'rxjs';
+import { catchError, exhaustMap, map } from 'rxjs/operators';
+import {
+  LoginPageActions,
+  AuthApiActions,
+} from '../actions';
+import { Credentials } from '../models/user';
+import { AuthService } from '../services/auth.service';
+
+@Injectable()
+export class AuthEffects {
+  login$ = createEffect(() =>
+    this.actions$.pipe(
+      // Filters by Action Creator 'login'
+      ofType(LoginPageActions.login),
+      exhaustMap(action =>
+        this.authService.login(action.credentials).pipe(
+          map(user => AuthApiActions.loginSuccess({ user })),
+          catchError(error => of(AuthApiActions.loginFailure({ error })))
+        )
+      )
+    )
+  );
+
+  constructor(
+    private actions$: Actions,
+    private authService: AuthService
+  ) {}
+}
+</code-example>
+
+## `mapToAction`
+
+Many effects are used to call APIs and due to the nature of network communication 
+some of them may fail. That means that the service call should ideally be wrapped
+with `catchError` to transform the failed request into another Action.
+
+Not only `catchError` is necessary, it's has to be used before the stream is 
+flattened or has to be constructed to re-subscribe to the source Observable.
+Missing such error handling results in bugs that are hard to discover during PR 
```

Comment:
I think we can leave it out `during PR submittion.` here

Reply:
Sg